apis/kubedb/v1alpha1/mongodb_helpers.go

```diff
@@ -416,19 +416,18 @@ func (m *MongoDBSpec) setDefaultProbes(podTemplate *ofst.PodTemplateSpec) {
 		return
 	}
 
-	cmd := []string{
-		"mongo",
-		"--host=localhost",
-		"--eval",
-		"db.adminCommand('ping')",
+	var sslArgs string
+	if m.SSLMode == SSLModeRequireSSL {
+		sslArgs = fmt.Sprintf("--ssl --sslCAFile=/data/configdb/%v --sslPEMKeyFile=/data/configdb/%v", MongoTLSCertFileName, MongoClientPemFileName)
 	}
 
-	if m.SSLMode == SSLModeRequireSSL {
-		cmd = append(cmd, []string{
-			"--ssl",
-			fmt.Sprintf("--sslCAFile=/data/configdb/%v", MongoTLSCertFileName),
-			fmt.Sprintf("--sslPEMKeyFile=/data/configdb/%v", MongoClientPemFileName),
-		}...)
+	cmd := []string{
+		"bash",
+		"-c",
+		fmt.Sprintf(`if [[ $(mongo admin --host=localhost %v --username=$MONGO_INITDB_ROOT_USERNAME --password=$MONGO_INITDB_ROOT_PASSWORD --authenticationDatabase=admin --quiet --eval "db.adminCommand('ping').ok" ) -eq "1" ]]; then 
```

Comment:
so we don't need version based check?

Reply:
`--eval` will be removed from `4.2`, which is not released yet. So, we don't need version based checking yet.

Reply:
Ok. Can you file a bug so that we can come back to it?